protocols/perf/Cargo.toml

```diff
@@ -13,7 +13,7 @@ categories = ["network-programming", "asynchronous"]
 [dependencies]
 anyhow = "1"
 async-std = { version = "1.9.0", features = ["attributes"] }
-clap = { version = "4.2.1", features = ["derive"] }
+clap = { version = "4.2.4", features = ["derive"] }
```

Comment:
This being part of the regular dependencies adds a lot of code for users that activate the perf feature.

It is for testing only but I think I'd still feel better if these dependencies wouldn't be part of our normal dependency tree.

Not sure if we've previously discussed this but perhaps we should remove `perf` from our optional `libp2p` dependencies such that it doesn't show up as part of `full`?

cc @mxinden

Reply:
Discussed it here https://github.com/libp2p/rust-libp2p/pull/3508#discussion_r1125546920, though from a different perspective.

I don't think it is worth the inconsistency it introduces. Users should not be surprised that compile times are high when they use `full`. In terms of CI speed, we would need to build `clap` either way.

Reply:
All fair points.